### PR TITLE
Add gpu field

### PR DESCRIPTION
**Pull request by REestrella**

### CODE REVIEW

**Comment by REestrella** [comment]
I ran the script and everything works fine. Just one region(us-east-1-msp-1) failed.
```
Failed: invalid character '<' looking for beginning of value
```
I did not find the reason. /cc @mrhello369 for help

**Comment by mrhello369** [comment]
> I ran the script and everything works fine. Just one region(us-east-1-msp-1) failed.
> 
> ```
> Failed: invalid character '<' looking for beginning of value
> ```
> 
> I did not find the reason. /cc @mrhello369 for help

Let me check

File: hack/tools/gendata.go [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+	"io"
+	"net/http"
+	"os"
+	"strings"
+
+	"kubefin.dev/go-tools/awspricing"
+)
+
+// Usage:
+// go build -o awspricing/gendata hack/tools/gendata.go
```
Comment (mrhello369) [inline hack/tools/gendata.go]: How about just use `go run hack/tools/gendata.go`

File: hack/tools/gendata.go [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+				regionData[item.Region] = append(regionData[item.Region], item)
+			}
+
+			for region, data := range regionData {
+				jsonRet, err := json.Marshal(data)
+				if err != nil {
+					return err
+				}
+
+				err = os.WriteFile("data/"+region+".json", jsonRet, 0644)
```
Comment (mrhello369) [inline hack/tools/gendata.go]: If with `go run hack/tools/gendata.go`, here should be `err = os.WriteFile("awspricing/data/"+region+".json", jsonRet, 0644)`

File: hack/tools/gendata.go [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+				if err != nil {
+					return err
+				}
+			}
+
+			return nil
+		}
+
+		if err := getDataFunc(); err != nil {
```
Comment (mrhello369) [inline hack/tools/gendata.go]: There is QPS limitations, lets try it multiple times.
```
		for i := 0; i < 6; i++ {
			if err := getDataFunc(); err != nil {
				klog.Errorf("Failed:%v", err)
				time.Sleep(time.Second * 30)
				continue
			}
			break
		}
```

File: hack/tools/gendata.go [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+			for region, data := range regionData {
+				jsonRet, err := json.Marshal(data)
+				if err != nil {
+					return err
+				}
+
+				err = os.WriteFile("data/"+region+".json", jsonRet, 0644)
+				if err != nil {
+					return err
+				}
```
Comment (mrhello369) [inline hack/tools/gendata.go:84]: Log out the success information would be better:
```go
klog.Infof("EC2 price data(%s) download/parse finished....", region)
```

**Comment by mrhello369** [comment]
> I ran the script and everything works fine. Just one region(us-east-1-msp-1) failed.
> 
> ```
> Failed: invalid character '<' looking for beginning of value
> ```
> 
> I did not find the reason. /cc @mrhello369 for help

I find it, it's because there is a blank space in types.go, `"us-east-1-msp-1 "` should be `"us-east-1-msp-1"`

**Review by mrhello369 — APPROVED** [review]
/lgtm

**Pull request merged**